### PR TITLE
keep_keys: Add complex integration tests.

### PR DESCRIPTION
**Pull request by vbotka**
##### SUMMARY
Add integration tests.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
ansible.utils.keep_keys

##### ADDITIONAL INFORMATION
No additional info atm.
```

### CODE REVIEW

File: tests/integration/inventory [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[testgroup]
+testhost ansible_connection="local" ansible_pipelining="yes" ansible_python_interpreter="/home/roverflow/workspace/virtualvenvs/a216p312/bin/python"
```
Comment (KB-perByte) [inline tests/integration/inventory:2]: Change this please the interpreter path, I don't think we need this file

File: tests/integration/targets/utils_keep_keys/tasks/complex1.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      ansible.builtin.debug:
+        var: result|to_yaml
+      when: debug_test|d(false)|bool
+    - name: Test keep_keys assert
+      ansible.builtin.assert:
+        that: result == target
+
+- name: Test map keep_keys
+  vars:
+    result: "{{ l2 | map('ansible.utils.keep_keys', target=['p1', 'p2'])| list }}"
```
Comment (KB-perByte) [inline tests/integration/targets/utils_keep_keys/tasks/complex1.yaml:21]: fix this lint warning too!

File: tests/integration/targets/utils_keep_keys/tasks/complex1.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      ansible.builtin.debug:
+        var: result|to_yaml
+      when: debug_test|d(false)|bool
+    - name: Test keep_keys end_with '2' assert
+      ansible.builtin.assert:
+        that: result == target
+
+- name: Test keep_keys regex
+  vars:
+    result: "{{ l1 | ansible.utils.keep_keys(target=['p1', 'p2'],matching_parameter= 'regex') }}"
```
Comment (KB-perByte) [inline tests/integration/targets/utils_keep_keys/tasks/complex1.yaml:112]: same with all the warning

File: tests/integration/targets/utils_keep_keys/tasks/complex1.yaml [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+      ansible.builtin.debug:
+        var: result|to_yaml
+      when: debug_test|d(false)|bool
+    - name: Test keep_keys end_with assert
+      ansible.builtin.assert:
+        that: result == target
+
+- name: Test keep_keys ends_with '2'
+  vars:
+    result: "{{ l1 | ansible.utils.keep_keys(target=['2'],matching_parameter= 'ends_with') }}"
```
Comment (KB-perByte) [inline tests/integration/targets/utils_keep_keys/tasks/complex1.yaml:95]: same

File: tests/integration/targets/utils_keep_keys/tasks/complex1.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      ansible.builtin.debug:
+        var: result|to_yaml
+      when: debug_test|d(false)|bool
+    - name: Test keep_keys starts_with 'p' assert
+      ansible.builtin.assert:
+        that: result == target
+
+- name: Test keep_keys ends_with
+  vars:
+    result: "{{ l1 | ansible.utils.keep_keys(target=['p1', 'p2'],matching_parameter= 'ends_with') }}"
```
Comment (KB-perByte) [inline tests/integration/targets/utils_keep_keys/tasks/complex1.yaml:78]: same

File: tests/integration/targets/utils_keep_keys/tasks/complex1.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+      ansible.builtin.debug:
+        var: result|to_yaml
+      when: debug_test|d(false)|bool
+    - name: Test keep_keys starts_with assert
+      ansible.builtin.assert:
+        that: result == target
+
+- name: Test keep_keys starts_with 'p'
+  vars:
+    result: "{{ l1 | ansible.utils.keep_keys(target=['p'],matching_parameter= 'starts_with') }}"
```
Comment (KB-perByte) [inline tests/integration/targets/utils_keep_keys/tasks/complex1.yaml:61]: same

File: tests/integration/targets/utils_keep_keys/tasks/complex1.yaml [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+      ansible.builtin.debug:
+        var: result|to_yaml
+      when: debug_test|d(false)|bool
+    - name: Test map keep_keys assert
+      ansible.builtin.assert:
+        that: result == target
+
+- name: Test keep_keys starts_with
+  vars:
+    result: "{{ l1 | ansible.utils.keep_keys(target=['p1', 'p2'],matching_parameter= 'starts_with') }}"
```
Comment (KB-perByte) [inline tests/integration/targets/utils_keep_keys/tasks/complex1.yaml:44]: same

**Review by KB-perByte — CHANGES_REQUESTED** [review]

File: tests/integration/targets/utils_keep_keys/vars/main.yaml [hunk 8]
```diff
@@ -22,3 +22,22 @@ keep_default:
           is_enabled: true
     - interface_name: eth2
       is_enabled: false
+
+l1:
```
Comment (Ruchip16) [inline tests/integration/targets/utils_keep_keys/vars/main.yaml:26]: @roverflow can you please create a new file called complex.yaml which has the output of all the complex integration tests, as the format of our and these tests are different its better if we keep them in a different file, you can refer this [PR](https://github.com/ansible-collections/ansible.utils/pull/200) for that.
CC: @NilashishC

**Review by Ruchip16 — CHANGES_REQUESTED** [review]

File: tests/integration/targets/utils_keep_keys/vars/main.yaml (hunk 8 above)
Comment (NilashishC) [reply tests/integration/targets/utils_keep_keys/vars/main.yaml:26]: +1 to that.

**Review by KB-perByte — APPROVED** [review]

**Review by Ruchip16 — APPROVED** [review]

**Pull request merged**